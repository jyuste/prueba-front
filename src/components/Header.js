import React from 'react';

import SearchBox from './SearchBox';
import logo from '../assets/img/github-512.png';

const Header = (props) => {

    return(
        <nav className="navbar is-black mb-5" role="navigation" aria-label="main navigation">
          <div className="navbar-brand">
            <div className="navbar-item logo-container">
              <img src={logo} alt="logo" />
            
            </div>
          </div>
              <div className="navbar-item">
                <form onSubmit={props.handler}>
                  <SearchBox />
                </form>
              </div>
        </nav>
    )
}

export default Header