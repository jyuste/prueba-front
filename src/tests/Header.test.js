import { React } from "react";

import { shallow } from "enzyme";
import Header from "../components/Header";

describe('Pruebas en <Header />', () => {

    const wrapper = shallow(<Header />)
    
    test('El componente se crea correctamente', () => {
        expect( wrapper ).toMatchSnapshot();
    })

    test('Logo tiene atributos definidos', () => {
        const logo = wrapper.find('img')
        expect(logo.prop('src')).toBeDefined;    
        expect(logo.prop('alt')).toBe('logo');
    })
})