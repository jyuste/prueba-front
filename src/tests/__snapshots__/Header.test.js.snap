// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Header /> El componente se crea correctamente 1`] = `
<nav
  aria-label="main navigation"
  className="navbar is-black mb-5"
  role="navigation"
>
  <div
    className="navbar-brand"
  >
    <div
      className="navbar-item logo-container"
    >
      <img
        alt="logo"
        src="github-512.png"
      />
    </div>
  </div>
  <div
    className="navbar-item"
  >
    <form>
      <SearchBox />
    </form>
  </div>
</nav>
`;
